server/tailsql/options.go

```diff
@@ -551,12 +558,22 @@ func (d *DBSpec) countFields() (n int) {
 }
 
 func (d *DBSpec) checkValid() error {
-	switch {
-	case d.Source == "":
-		return errors.New("missing source")
-	case d.Driver == "":
+	if d.Source == "" {
+		return errors.New("missing source name")
+	}
+
+	// Case 1: A programmatic data source.
+	if d.DB != nil {
+		if d.countFields() != 0 {
+			return errors.New("exactly one connection source must be set")
```

Comment:
zero

Reply:
Sorry, I don't understand what you mean here.

Reply:
Oh it's checking if the number of fields is 0 but saying it needs 1.

Reply:
Oh! 🤦🏻‍♀️ Good catch, thanks 😅